app/models/user.rb

```diff
@@ -9,4 +9,39 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.compact
+  end
+
+  # Users who have yet to confirm friend requests
+  def friend_request_sent
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
```

Comment:
**Optional**: It's always better to create scopes instead of methods like this to query the DB. 
You can create one that looks for all friendships `where` the friendship is not confirmed.